From f5fb20e03751d9c84d046f748a3c2fc58bad4eda Mon Sep 17 00:00:00 2001
From: Ramya P Karanth <ramya.p.karanth@intel.com>
Date: Tue, 27 Oct 2020 20:20:21 +0530
Subject: [PATCH 1/1] pwm-dwc writel

Fix the dwc_writel params in runtime suspend and resume function

Signed-off-by: Ramya P Karanth <ramya.p.karanth@intel.com>
---
 drivers/pwm/pwm-dwc.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/pwm/pwm-dwc.c b/drivers/pwm/pwm-dwc.c
index 2defe8fea0d0..cdb83aa8717c 100644
--- a/drivers/pwm/pwm-dwc.c
+++ b/drivers/pwm/pwm-dwc.c
@@ -337,14 +337,14 @@ static int dwc_pci_runtime_suspend(struct device *dev)
 	j1 = j0 + delay;
 
 	cgsr_reg = dwc_readl(dwc->base, DWC_TIM_CGSR);
-	dwc_writel(dwc->base, DWC_TIM_CGSR_CG, DWC_TIM_CGSR);
+	dwc_writel(dwc->base, DWC_TIM_CGSR, DWC_TIM_CGSR_CG);
 
 	d0i3c_reg = dwc_readl(dwc->base, DWC_TIM_D0I3C);
 
 	if (d0i3c_reg & DWC_TIM_D0I3_CIP) {
 		dev_info(dev, "%s d0i3c CIP detected", __func__);
 	} else {
-		dwc_writel(dwc->base, DWC_TIM_D0I3_EN, DWC_TIM_D0I3C);
+		dwc_writel(dwc->base, DWC_TIM_D0I3C, DWC_TIM_D0I3_EN);
 		d0i3c_reg = dwc_readl(dwc->base, DWC_TIM_D0I3C);
 	}
 
@@ -376,7 +376,7 @@ static int dwc_pci_runtime_resume(struct device *dev)
 	cgsr_reg = dwc_readl(dwc->base, DWC_TIM_CGSR);
 
 	if (cgsr_reg & DWC_TIM_CGSR_CG)
-		dwc_writel(dwc->base, (cgsr_reg & ~DWC_TIM_CGSR_CG), DWC_TIM_CGSR);
+		dwc_writel(dwc->base, DWC_TIM_CGSR, (cgsr_reg & ~DWC_TIM_CGSR_CG));
 
 	d0i3c_reg = dwc_readl(dwc->base, DWC_TIM_D0I3C);
 
@@ -390,7 +390,7 @@ static int dwc_pci_runtime_resume(struct device *dev)
 		if (d0i3c_reg & DWC_TIM_D0I3_RR)
 			d0i3c_reg |= DWC_TIM_D0I3_RR;
 
-		dwc_writel(dwc->base, d0i3c_reg, DWC_TIM_D0I3C);
+		dwc_writel(dwc->base, DWC_TIM_D0I3C, d0i3c_reg);
 		d0i3c_reg = dwc_readl(dwc->base, DWC_TIM_D0I3C);
 	}
 
-- 
2.17.1

